sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py

```diff
@@ -0,0 +1,163 @@
+    LOGGING_EXPORT_INTERVAL_MS_ARG,
+    SAMPLING_RATIO_ARG,
+)
+from azure.monitor.opentelemetry._types import ConfigurationValue
+from azure.monitor.opentelemetry._vendor.v0_38b0.opentelemetry.instrumentation.dependencies import (
+    get_dependency_conflicts,
+)
+from azure.monitor.opentelemetry._vendor.v0_38b0.opentelemetry.instrumentation.instrumentor import (
+    BaseInstrumentor,
+)
+from azure.monitor.opentelemetry.exporter import (
+    ApplicationInsightsSampler,
+    AzureMonitorLogExporter,
+    AzureMonitorMetricExporter,
+    AzureMonitorTraceExporter,
+)
+from azure.monitor.opentelemetry.util._configurations import _get_configurations
+
+_logger = getLogger(__name__)
+
+
+_SUPPORTED_INSTRUMENTED_LIBRARIES_DEPENDENCIES_MAP = {
+    "django": ("django >= 1.10",),
+    "fastapi": ("fastapi ~= 0.58",),
+    "flask": ("flask >= 1.0, < 3.0",),
+    "psycopg2": ("psycopg2 >= 2.7.3.1",),
+    "requests": ("requests ~= 2.0",),
+    "urllib": tuple(),
+    "urllib3": ("urllib3 >= 1.0.0, < 2.0.0",),
+}
+
+
+def configure_azure_monitor(**kwargs) -> None:
+    """
+    This function works as a configuration layer that allows the
+    end user to configure OpenTelemetry and Azure monitor components. The
+    configuration can be done via arguments passed to this function.
+    :keyword str connection_string: Connection string for your Application Insights resource.
+    :keyword ManagedIdentityCredential/ClientSecretCredential credential: Token credential, such as
+    ManagedIdentityCredential or ClientSecretCredential, used for Azure Active Directory (AAD) authentication. Defaults
+    to None.
+    :keyword bool disable_offline_storage: Boolean value to determine whether to disable storing failed telemetry
+    records for retry. Defaults to `False`.
+    :keyword str storage_directory: Storage directory in which to store retry files. Defaults to
+    `<tempfile.gettempdir()>/Microsoft/AzureMonitor/opentelemetry-python-<your-instrumentation-key>`.
+    :rtype: None
+    """
+
+    configurations = _get_configurations(**kwargs)
+
+    disable_tracing = configurations[DISABLE_TRACING_ARG]
+    disable_logging = configurations[DISABLE_LOGGING_ARG]
+    disable_metrics = configurations[DISABLE_METRICS_ARG]
+
+    # Setup tracing pipeline
+    if not disable_tracing:
+        _setup_tracing(configurations)
+
+    # Setup logging pipeline
+    if not disable_logging:
+        _setup_logging(configurations)
+
+    # Setup metrics pipeline
+    if not disable_metrics:
+        _setup_metrics(configurations)
+
+    # Setup instrumentations
+    # Instrumentations need to be setup last so to use the global providers
+    # instanstiated in the other setup steps
+    _setup_instrumentations()
+
+
+def _setup_tracing(configurations: Dict[str, ConfigurationValue]):
+    sampling_ratio = configurations[SAMPLING_RATIO_ARG]
+    tracer_provider = TracerProvider(
+        sampler=ApplicationInsightsSampler(sampling_ratio=sampling_ratio),
```

Comment:
More typing fun: sampling_ratio is expected to be a float, so can maybe use `typing.cast` here: `sampling_ratio=cast(float, sampling_ratio)`

Reply:
done